### PR TITLE
db.get_torrent()

### PR DESCRIPTION
**Pull request by virtualhealthcitizen**
First pass at implementing `db.get_torrent()`, `db.add_plugin()`, and a testing module (see `test_db.py`).

### CODE REVIEW

**Comment by badstreff** [comment]
I switched the base to the dev branch.

**Comment by badstreff** [comment]
This looks way better, I'm adding some comments for things that will need changed.

File: debug/db.py [hunk 1]
```diff
@@ -239,18 +239,77 @@ def get_torrent(self, info_hash):
...
+        comment TEXT,
+        created_by TEXT,
+        creation_time TIMESTAMP,
+        piece_length INT,
+        pieces BYTEA
+        """
+        
+        cursor = self._connection.cursor()
+        try:
+            cursor.execute( ("SELECT * FROM torrents WHERE info_hash = 'asdfasdf'") )
```
Comment (badstreff) [inline debug/db.py]: 'asdfasdf' here should be replaced with the info_hash parameter passed to the function

File: debug/db.py [hunk 2]
```diff
@@ -239,18 +239,77 @@ def get_torrent(self, info_hash):
...
+        torrent[b'created by'] = str.encode(created_by)
+        torrent[b'creation time'] = creation_time
+        torrent[b'piece length'] = piece_length
+        torrent[b'pieces'] = pieces
+        torrent[b'files'] = []
+        torrent[b'info'] = { 
+                             b'name': str.encode(name),
+                             b'piece length': piece_length,
+                             b'pieces': pieces,
+                             b'files': [
```
Comment (badstreff) [inline debug/db.py]: You will need to make a query to the files table to populate this list.

File: debug/db.py [hunk 3]
```diff
@@ -239,18 +239,77 @@ def get_torrent(self, info_hash):
...
+        
+        info_hash = bytes(tup[0])
+        name = tup[1]
+        comment = tup[2]
+        created_by = tup[3]
+        creation_time = tup[4]
+        piece_length = tup[5]
+        pieces = bytes(tup[6])
+
+        torrent = {}
```
Comment (badstreff) [inline debug/db.py:258]: You are missing the 'announcer-lisr' key, another call will need made to the announcers table to get this information

File: debug/db.py [hunk 4]
```diff
@@ -239,18 +239,77 @@ def get_torrent(self, info_hash):
...
+        torrent[b'info hash'] = info_hash
+        torrent[b'name'] = str.encode(name)
+        torrent[b'comment'] = str.encode(comment)
+        torrent[b'created by'] = str.encode(created_by)
+        torrent[b'creation time'] = creation_time
+        torrent[b'piece length'] = piece_length
+        torrent[b'pieces'] = pieces
+        torrent[b'files'] = []
```
Comment (badstreff) [inline debug/db.py]: This line isn't needed as the 'files' key is not used outside of the info key

File: debug/db.py [hunk 5]
```diff
@@ -239,18 +239,77 @@ def get_torrent(self, info_hash):
...
         Returns:
             BOOL: success or failure
         """
-        pass
+        
+        cursor = self._connection.cursor()
+        try:
+            cursor.execute( ("INSERT INTO plugins VALUES (%s, %s, %s) "
+                             "ON CONFLICT (id) DO NOTHING"),
+                             ('001', 'some/url/and/some/goats.txt', datetime.datetime.now()) )
```
Comment (badstreff) [inline debug/db.py]: There is no 'id' column in the plugin table, the URL is the Primary Key, so use that when checking for conflicts.
Additionally, 'some/url/and/some/goats.txt' should be the url passed to the function.

Comment (virtualhealthcitizen) [reply debug/db.py]: I got rid of the `id` and now use the `url` to check for conflicts.
Currently working on finalizing fixes for this and a few other issues.

File: debug/db.py (hunk 4 above)
Comment (virtualhealthcitizen) [reply debug/db.py]: I got rid of that line last night.

-- snek

File: debug/db.py (hunk 3 above)
Comment (virtualhealthcitizen) [reply debug/db.py:258]: Noted.

File: debug/db.py (hunk 2 above)
Comment (virtualhealthcitizen) [reply debug/db.py]: Is a work in progress.

File: debug/db.py (hunk 1 above)
Comment (virtualhealthcitizen) [reply debug/db.py]: Done, just haven't merged my fork with upstream yet.

**Pull request closed without merging**